Hotsapi.Uploader.Common/Analyzer.cs

```diff
@@ -22,7 +22,10 @@ public class Analyzer : IAnalyzer
         public Replay Analyze(ReplayFile file)
         {
             try {
-                var result = DataParser.ParseReplay(file.Filename, false, false, false, true);
+                var parseOptions = new ParseOptions() {
+                    AllowPTR = true

```

Comment:
So I switched up the parse options from feedback provided by barrret before he merged your changes.  So just wanted to make sure you saw those changes.  Also, why allowPTR?  The uploader doesn’t upload PTR games as far as I am aware

Reply:
The again, you have to parse PTR to know it’s PTR.  So probably why.

Reply:
Indeed, that doesn't look right. The old method signature was `ParseReplay(string fileName, bool ignoreErrors, bool deleteFile, bool allowPTRRegion = false, bool skipEventParsing = false, bool skipUnitParsing = false, bool skipMouseMoveEvents = false, bool detailedBattleLobbyParsing = false)`  so that's `ignoreErrors = false`, `deleteFile = false`, `allowPTR = false`, `skipEventParseing = true`, `skipUnitParsing=false`, `skipMouseMoveEvents = false` and `detailedBattleLobbyParsing = false`. But that's odd: why not skip mouse events. Closing this to look again.

Reply:
Could you maybe help me out with the versioning? I'm completely lost on where to find which commit is published in which version of the package.

Reply:
The owner of the repo requested the defaulted state be the same as the previous default state for consistently and reduce confusion.  Which makes sense to me

Reply:
The default state for which method in which version of the source?

Reply:
You can take a look at the conversation here https://github.com/barrett777/Heroes.ReplayParser/pull/91

What I ended up doing was taking out TypicalParsing and replacing it with DefaultParsing.  Then I added MediumParsing that should represent the previous TypicalParsing.

Take a look at that master repo for what the nuget package is defined as.  If you think the options are messed up, you can throw a new one in there and do a pull request directly to that master.

Reply:
It's a bit difficult to see what changed, since the version that the uploader was using previously isn't in hotsapi/master, and the commits don't seem to have preserved history. I tried to figure it out locally, but I broke something, it seems. I definitely we don't think another level apart from minimal, full, default, typical, medium, and the defaults set in the constructor. To be honest, IMO the distinction between medium, typical and default is pretty unclear already.

Reply:
- DefaultParsing - This is what is set by the class be default, and matches what previously set by multiple options sent through the constructor
- MinimalParsing - This is equivalent to what you had created.  It skips most parsing
- MediumParsing - This one is the one you want to use I believe.  It has the options set for improved parsing speed, while still keeping the data we need for HotsApi.
- FullParsing - self-explanatory

Current Options
https://github.com/barrett777/Heroes.ReplayParser/blob/master/Heroes.ReplayParser/ParseOptions.cs

Look at this code to see previous options
https://github.com/barrett777/Heroes.ReplayParser/tree/16218b086e86ddf1c2a89bbb5f68a5f0abf03a77

Reply:
For some reason, at the moment parsing is returning exception when called from the nuget package. Unfortunately, it doesn't provide more information than that, and I can't reproduce on HEAD of master of Barrett's repo. Do you happen to know at which commit version 1.2.3 of the nuget package was published?

Reply:
https://github.com/barrett777/Heroes.ReplayParser/releases

Should be the commit behind the latest pull
https://github.com/barrett777/Heroes.ReplayParser/tree/9dda6eed3c322d1270263c904b5765b277052828

If the latest pull is working, you may want to get @barrett777 to push the latest request to nuget

Reply:
Unfortunately, I can't get the indicated version to build.

Reply:
Ya I can't either.  Interesting.  Can you open an issue up on the main repo?

Reply:
I did push the nuget package into my lambda parser and it worked without issue

Reply:
I opened https://github.com/barrett777/Heroes.ReplayParser/issues/97

Reply:
I have the impression that the build error is related to GitVersion. I asked around at https://gitter.im/GitTools/GitVersion?at=5e29c4d2f85dba0aab003034

Reply:
@martijnhoekstra GitVersion is a development-only dependency.  It shouldn't be included in or affect the NuGet package: https://github.com/barrett777/Heroes.ReplayParser/blob/master/Heroes.ReplayParser/Heroes.ReplayParser.csproj#L15-L16

Reply:
I did have issues in development with replays all returning Exception, and I think the issue was different versions of 'sharpcompress'

Reply:
A version mismatch of sharpcompress was indeed the issue. It's troublesome that exceptions are swallowed like this to debug this.